File: Dummy/hawkshaw/library/datalayer/models/Timestamp.java
package com.hawkshaw.library.datalayer.models;

import B3.f;
import kotlinx.serialization.KSerializer;

@f(with = TimestampSerializer.class)
public final class Timestamp {
    public static final Companion Companion = new Companion((j3.f) null);
    private final long milliseconds;

    public static final class Companion {
        private Companion() {
        }

        public /* synthetic */ Companion(j3.f fVar) {
            this();
        }

        public final KSerializer serializer() {
            return TimestampSerializer.INSTANCE;
        }
    }

    public Timestamp() {
        this(0, 1, (j3.f) null);
    }

    public static /* synthetic */ void getMilliseconds$annotations() {
    }

    public final long getMilliseconds() {
        return this.milliseconds;
    }

    /* JADX INFO: this call moved to the top of the method (can break code semantics) */
    public /* synthetic */ Timestamp(long j5, int i5, j3.f fVar) {
        this((i5 & 1) != 0 ? 0 : j5);
    }

    public Timestamp(long j5) {
        this.milliseconds = j5;
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/models/Timestamp.kt
package com.hawkshaw.library.datalayer.models

import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.descriptors.PrimitiveKind
import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
import kotlinx.serialization.descriptors.SerialDescriptor
import kotlinx.serialization.encoding.Decoder
import kotlinx.serialization.encoding.Encoder
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

/**
 * Timestamp representation for serialization purposes
 */
@Serializable(with = TimestampSerializer::class)
data class Timestamp(val milliseconds: Long = 0)

/**
 * Serializer for Timestamp class that converts between ISO RFC3339 format strings and Timestamp objects
 */
object TimestampSerializer : KSerializer<Timestamp> {
    private val rfc3339Format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX", Locale.US).apply {
        timeZone = TimeZone.getTimeZone("UTC")
    }
    
    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("Timestamp", PrimitiveKind.STRING)
    
    override fun deserialize(decoder: Decoder): Timestamp {
        val dateString = decoder.decodeString()
        return Timestamp(rfc3339Format.parse(dateString)?.time ?: 0)
    }
    
    override fun serialize(encoder: Encoder, value: Timestamp) {
        val dateString = rfc3339Format.format(Date(value.milliseconds))
        encoder.encodeString(dateString)
    }
} 